source/module_elecstate/potentials/H_TDDFT_pw.cpp

```diff
@@ -11,147 +14,282 @@ int H_TDDFT_pw::istep = -1;
-            }
-            else if (j >= this->rho_basis_->nx * 0.05 && j < this->rho_basis_->nx * 0.95)
-            {
-                vextold[ir] = -0.0019447 * j / this->rho_basis_->nx * this->ucell_->lat0;
-            }
-            else if (j >= this->rho_basis_->nx * 0.95)
-            {
-                vextold[ir] = (0.019447 * (1.0 * j / this->rho_basis_->nx - 1) - 0.001069585) * this->ucell_->lat0;
-            }
-        }
-        else if (ELEC_evolve::td_vext_dire == 3)
-        {
-            if (i < this->rho_basis_->nx * 0.05)
-            {
-                vextold[ir] = (0.019447 * i / this->rho_basis_->nx - 0.001069585) * this->ucell_->lat0;
-            }
-            else if (i >= this->rho_basis_->nx * 0.05 && i < this->rho_basis_->nx * 0.95)
-            {
-                vextold[ir] = -0.0019447 * i / this->rho_basis_->nx * this->ucell_->lat0;
-            }
-            else if (i >= this->rho_basis_->nx * 0.95)
-            {
-                vextold[ir] = (0.019447 * (1.0 * i / this->rho_basis_->nx - 1) - 0.001069585) * this->ucell_->lat0;
-            }
-        }
+    delete[] vext_space;
 
-        // Gauss
-        if (ELEC_evolve::td_vexttype == 1)
-        {
-            const double w = 22.13; // eV
-            const double sigmasquare = 700;
-            const double timecenter = 700;
-            // Notice: these three parameters should be written in INPUT. I will correct soon.
-            const double timenow
-                = (H_TDDFT_pw::istep - timecenter) * ELEC_evolve::td_scf_thr * 41.34; // 41.34 is conversion factor of fs-a.u.
-            vext[ir] = vextold[ir] * cos(w / 27.2116 * timenow) * exp(-timenow * timenow * 0.5 / (sigmasquare))
-                        * 0.25; // 0.1 is modified in 2018/1/12
-        }
+    ModuleBase::timer::tick("H_TDDFT_pw", "cal_fixed_v");
+    return;
+}
 
-        // HHG of H atom
-        if (ELEC_evolve::td_vexttype == 2)
-        {
-            const double w_h = 0.0588; // a.u.
-            const int stepcut1 = 1875;
-            const int stepcut2 = 5625;
-            const int stepcut3 = 7500;
-            // The parameters should be written in INPUT!
-            if (H_TDDFT_pw::istep < stepcut1)
-            {
-                vext[ir] = vextold[ir] * 2.74 * H_TDDFT_pw::istep / stepcut1
-                            * cos(w_h * H_TDDFT_pw::istep * ELEC_evolve::td_scf_thr * 41.34); // 2.74 is equal to E0;
-            }
-            else if (H_TDDFT_pw::istep < stepcut2)
-            {
-                vext[ir] = vextold[ir] * 2.74 * cos(w_h * H_TDDFT_pw::istep * ELEC_evolve::td_scf_thr * 41.34);
-            }
-            else if (H_TDDFT_pw::istep < stepcut3)
-            {
-                vext[ir] = vextold[ir] * 2.74 * (stepcut3 - H_TDDFT_pw::istep) / stepcut1
-                            * cos(w_h * H_TDDFT_pw::istep * ELEC_evolve::td_scf_thr * 41.34);
-            }
-        }
+void H_TDDFT_pw::read_parameters()
```

Comment:
@lyb9812 Please use "read_parameters(Input* in)" but not global class INPUT

Reply:
Ok.